--- config.sub	2020-07-10 19:59:02.000000000 +0800
+++ config.sub	2026-01-15 13:04:38.500307152 +0800
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -ohos*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
