--- build-aux/config.sub	2023-02-03 21:21:13.000000000 +0800
+++ build-aux/config.sub	2026-01-15 13:04:38.509307153 +0800
@@ -1758,7 +1758,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* )
+	     | fiwix* | mlibc* | ohos* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
@@ -1779,7 +1779,8 @@
 # (given a valid OS), if there is a kernel.
 case $kernel-$os in
 	linux-gnu* | linux-dietlibc* | linux-android* | linux-newlib* \
-		   | linux-musl* | linux-relibc* | linux-uclibc* | linux-mlibc* )
+		   | linux-musl* | linux-relibc* | linux-uclibc* | linux-mlibc* \
+			   | linux-ohos* )
 		;;
 	uclinux-uclibc* )
 		;;
--- libtextstyle/build-aux/config.sub	2023-02-03 21:21:42.000000000 +0800
+++ libtextstyle/build-aux/config.sub	2026-01-15 13:04:38.511307153 +0800
@@ -1758,7 +1758,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* )
+	     | fiwix* | mlibc* | ohos* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
@@ -1779,7 +1779,8 @@
 # (given a valid OS), if there is a kernel.
 case $kernel-$os in
 	linux-gnu* | linux-dietlibc* | linux-android* | linux-newlib* \
-		   | linux-musl* | linux-relibc* | linux-uclibc* | linux-mlibc* )
+		   | linux-musl* | linux-relibc* | linux-uclibc* | linux-mlibc* \
+			   | linux-ohos* )
 		;;
 	uclinux-uclibc* )
 		;;
--- gettext-tools/examples/hello-c++-kde/admin/config.sub	2019-05-11 19:29:32.000000000 +0800
+++ gettext-tools/examples/hello-c++-kde/admin/config.sub	2026-01-15 13:04:38.513307154 +0800
@@ -1223,6 +1223,8 @@
 	-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 		os=-mint
 		;;
+	-ohos*)
+		;;
 	-none)
 		;;
 	*)
